junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/process/EtcdProcess.java

```diff
@@ -0,0 +1,127 @@
+  private final int startupTimeout;
+  private File tempWalDir;
+  private File tempDataDir;
+
+  public EtcdProcess(BinaryManager binaryManager,
+      UnexpectedProcessStopHandler processStopHandler, boolean waitForHealthCheck,
+      int startupTimeout) {
+    this.binaryManager = binaryManager;
+    this.processStopHandler = processStopHandler;
+    this.waitForHealthCheck = waitForHealthCheck;
+    this.startupTimeout = startupTimeout;
+  }
+
+  public int startEtcd() {
+    try {
+      File etcdBinary = binaryManager.binaries().getEtcd();
+      tempWalDir = Files.createTempDirectory("etcdwal").toFile();
+      tempDataDir = Files.createTempDirectory("etcddata").toFile();
+      log.trace("Using temp wal dir: {} and temp data dir: {}", tempWalDir.getPath(),
+          tempDataDir.getPath());
+      var port = Utils.findFreePort();
+      var peerPort = Utils.findFreePort();
+
+      if (!etcdBinary.exists()) {
+        throw new KubeAPITestException(
+            "Missing binary for etcd on path: " + etcdBinary.getAbsolutePath());
+      }
+      etcdProcess = new ProcessBuilder(etcdBinary.getAbsolutePath(),
+          "-data-dir", tempDataDir.getPath(),
+          "-wal-dir", tempWalDir.getPath(),
+          "--listen-client-urls", "http://0.0.0.0:" + port,
+          "--advertise-client-urls", "http://0.0.0.0:" + port,
+          // the below added because of stability
+          "--initial-cluster", "default=http://localhost:" + peerPort,
+          "--initial-advertise-peer-urls", "http://localhost:" + peerPort,
+          "--listen-peer-urls", "http://localhost:" + peerPort)
+          .start();
+      Utils.redirectProcessOutputToLogger(etcdProcess.getInputStream(), etcdLog);
+      Utils.redirectProcessOutputToLogger(etcdProcess.getErrorStream(), etcdLog);
+      etcdProcess.onExit().thenApply(p -> {
+        if (!stopped) {
+          stopped = true;
+          log.error("etcd process stopped unexpectedly");
+          processStopHandler.processStopped(p);
+        }
+        return null;
+      });
+      log.debug("etcd started on port: {}", port);
+      if (waitForHealthCheck) {
+        waitUntilEtcdHealthy(port);
+      }
+      return port;
+    } catch (IOException e) {
+      throw new KubeAPITestException(e);
+    }
+  }
+
+  private void waitUntilEtcdHealthy(int port) {
+    new ProcessReadinessChecker().waitUntilReady(port, "health", "etcd", false, startupTimeout);
+  }
+
+  public void cleanEtcdData() {
+    try {
+      deleteDirectory(tempDataDir);
+      deleteDirectory(tempWalDir);
+    } catch (IOException e) {
+      throw new KubeAPITestException(e);
+    }
+  }
+
+  public void stopEtcd() {
+    if (stopped) {
+      return;
+    }
+    stopped = true;
+    if (etcdProcess != null) {
+      try {
+        etcdProcess.destroyForcibly();
+        etcdProcess.waitFor();
+      } catch (InterruptedException e) {
+        throw new KubeAPITestException(e);
+      }
+    }
+    log.debug("etcd stopped");
```

Comment:
This class seems to have some common logic as KubeAPIServerProcess, maybe it can be refactored to a common class.

Reply:
yes, can I make a separate PR for this?